src/libmeasurement_kit/ooni/dns_injection.cpp

```diff
@@ -16,9 +16,9 @@ void dns_injection(std::string input, Settings options, Callback<Var<Entry>> cb,
     Var<Entry> entry(new Entry);
     (*entry)["injected"] = nullptr;
     templates::dns_query(entry, "A", "IN", input, options["backend"],
-                         [=](Error, dns::Message message) {
+                         [=](Error, Var<dns::Message> message) {
                              logger->debug("dns_injection: got response");
-                             if (message.error_code == DNS_ERR_NONE) {
+                             if (message->error_code == DNS_ERR_NONE) {
```

Comment:
The quickest way to do that is probably to give a name to the `Error` function argument and make sure there is no error, otherwise don't attempt to access `message`, right?